cogs/annual_handover_and_reset.py

```diff
@@ -222,3 +227,109 @@ async def annual_roles_reset(self, ctx: TeXBotApplicationContext) -> None:
 
         logger.debug("Execution of reset roles command complete!")
         await initial_response.edit(content=":white_check_mark: Role reset complete!")
+
+class IterateYearChannelsCommandCog(TeXBotBaseCog):
+    """Cog class that defines the "/iterate-year-channels" command."""
+
+    @discord.slash_command(  # type: ignore[no-untyped-call, misc]
+        name="iterate-year-channels",
+        description="Iterates the year channels, archiving and creating channels as needed",
+    )
+    async def iterate_year_channels(self, ctx: TeXBotApplicationContext) -> None:
+        """
+        Definition and callback response of the "iterate-year-channels" command.
+
+        This command:
+        - Archives the current final-years channel
+        - Renames the current second year channel to final-years
+        - Renames the current first year channel to second-years
+        - Creates a new first years channel
+        """
+        main_guild: discord.Guild = self.bot.main_guild
```

Comment:
Missing shortcut accessor note. Refer to other commands to find the note to copy.